pkg/messaging/mqtt/pubsub_test.go

```diff
@@ -0,0 +1,437 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mqtt_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	mqtt_pubsub "github.com/mainflux/mainflux/pkg/messaging/mqtt"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	topic       = "topic"
+	chansPrefix = "channels"
+	channel     = "9b7b1b3f-b1b0-46a8-a717-b8213f9eda3b"
+	subtopic    = "engine"
+)
+
+var (
+	data = []byte("payload")
+)
+
+func TestPublisher(t *testing.T) {
+	msgChan := make(chan []byte)
+
+	// Subscribing with topic, and with subtopic, so that we can publish messages.
+	client, err := newClient(address, "clientID1", 30*time.Second)
```

Comment:
While we are at it, maybe move this timeout to constant as well, for better visibility/changability.

Reply:
Pushed